prover/crates/bin/prover_autoscaler/src/global/queuer.rs

```diff
@@ -2,31 +2,48 @@ use std::collections::HashMap;
 
 use anyhow::{Context, Ok};
 use reqwest::Method;
-use zksync_prover_job_monitor::autoscaler_queue_reporter::VersionedQueueReport;
+use zksync_config::configs::prover_autoscaler::QueueReportFields;
+use zksync_prover_job_monitor::autoscaler_queue_reporter::{QueueReport, VersionedQueueReport};
+use zksync_types::prover_dal::JobCountStatistics;
 use zksync_utils::http_with_retries::send_request_with_retries;
 
 use crate::metrics::{AUTOSCALER_METRICS, DEFAULT_ERROR_CODE};
 
 const MAX_RETRIES: usize = 5;
 
-#[derive(Debug)]
-pub struct Queue {
-    pub queue: HashMap<String, u64>,
-}
+pub type Queue = HashMap<(String, QueueReportFields), u64>;
 
 #[derive(Default)]
 pub struct Queuer {
     pub prover_job_monitor_url: String,
 }
 
+fn sum_queue(jobs: JobCountStatistics) -> u64 {
```

Comment:
I'd probably implemented this as method of `JobCountStatistics`

Reply:
Good idea, thx!
Implemented.